src/app/server/OnboardingCodesUtil.cpp

```diff
@@ -45,14 +47,17 @@ void PrintQRCode(chip::RendezvousInformationFlags rendezvousFlags)
             ChipLogProgress(AppServer, "Copy/paste the below URL in a browser to see the QR Code:");
             ChipLogProgress(AppServer, "%s?data=%s", qrCodeBaseUrl, &qrCodeBuffer[0]);
         }
+
+        ChipLogProgress(AppServer, "Manual pairing code: [%s]", manualPairingCode.c_str());
     }
     else
     {
         ChipLogError(AppServer, "Getting QR code failed!");
     }
 }
 
-CHIP_ERROR GetQRCode(std::string & QRCode, chip::RendezvousInformationFlags rendezvousFlags)
+CHIP_ERROR GetOnboardingCodes(std::string & QRCode, std::string & manualPairingCode,
```

Comment:
Personally, I would actually split it into two functions. There is even possible to cut off the manual pairing code or QR code part in future completely,  right? Also in some places, you use `(void) manualPairingCode` which might suggest that it is sometimes undesirable to have it.

However, function for printing is okay, as this is used for debugging.

Reply:
Yeah, I agree, that was my first thought. I created one method to avoid executing again all `ConfigurationMgr().Get` methods necessary to create setup payload for code generation purposes.

Reply:
Done